### DISCUSSION TITLE
why not use content-type in response header to auto set responseType in got?

### DISCUSSION THREAD

**Discussion by winterAndvelvet** [sindresorhus/got#1759] [Q&A] [answered]
#### What would you like to discuss?

responseType
Type: string
Default: 'text'

Note: When using streams, this option is ignored.

The parsing method. Can be 'text', 'json' or 'buffer'.

The promise also has .text(), .json() and .buffer() methods which return another Got promise for the parsed body.
It's like setting the options to {responseType: 'json', resolveBodyOnly: true} but without affecting the main Got promise.

so,why not use content-type in response header to auto set responseType ?
#### Checklist

- [x] I have read the documentation.

**Comment by sindresorhus** [comment, marked as answer]
auto-detecting `responseType` from `Content-Type` headers is a bad idea because:

1. servers lie constantly - many APIs return JSON but send `text/plain` or wrong content types. auto-parsing would break these.

2. security risk - malicious servers could send harmful content-type headers to trigger unexpected parsing behavior.

3. unpredictability - explicit is better. if you expect JSON, saying `responseType: 'json'` makes code clearer and avoids surprises.

4. error handling mess - what if content-type says JSON but body isnt valid JSON? adds complexity and weird failure modes.

the current explicit approach puts you in control, not the server. you decide how to parse, regardles of what headers claim.
